healthcheck/healthcheck.go

```diff
@@ -219,6 +266,15 @@ func (o *Observation) retryCheck(quit chan struct{},
 		var err error
 		select {
 		case err = <-o.Check():
+			// If our error is nil, we have passed our health check,
+			// so we'll invoke our success callback if defined and
+			// then exit.
+			if err == nil {
+				// Invoke the success callback.
+				o.OnSuccess()
```

Comment:
Think we should add a debug or log info here, so that we know lnd has tried to invoke the callback here, in case the callback site isn't working.

Reply:
Added!